ElectronClient/app/app.js

```diff
@@ -109,6 +110,18 @@ class Application extends BaseApplication {
 				newState.windowContentSize = action.size;
 				break;
 
+			case 'WINDOW_CONTENT_ZOOM_FACTOR_SET':
+
+				{
+					const MIN = 0.3;
+					const MAX = 3;
+					const zoomFactor = Math.max(MIN, Math.min(MAX, action.zoomFactor));
+					newState = Object.assign({}, state);
+					newState.windowContentZoomFactor = zoomFactor;
+					webFrame.setZoomFactor(zoomFactor);
```

Comment:
The reducer [should not perform any side effect](https://redux.js.org/basics/reducers#handling-actions) so please move the setZoomFactor call to `generalMiddleware()`. For example: `if (action.type === 'WINDOW_CONTENT_ZOOM_FACTOR_SET') webFrame.setZoomFactor(newState.windowContentZoomFactor);`